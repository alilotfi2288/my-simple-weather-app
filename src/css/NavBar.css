.nav {
  width: 100%;
  min-height: 10vh;
  background-color: #8E24AA;
  color: white;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding: 24px;
}

.nav h2 {
  font-weight: bold;
}

.nav .city-search {
  border: none;
  width: 30%;
  padding: 12px;
  border-radius: 24px;
  background-color: rgba(206, 147, 216, 0.5);
  color: white;
  text-align: center;
}

.nav .city-search::-webkit-input-placeholder {
  color: white;
}

.nav .city-search:-ms-input-placeholder {
  color: white;
}

.nav .city-search::-ms-input-placeholder {
  color: white;
}

.nav .city-search::placeholder {
  color: white;
}

.nav .city-search:focus {
  outline: thin white auto;
}

.nav .unit > * {
  padding: 0px 4px;
  font-weight: bold;
  cursor: pointer;
}

.nav .unit > *:nth-child(2) {
  cursor: auto;
}

.sugesstions {
  position: absolute;
  top: 9vh;
  left: 55%;
  -webkit-transform: translatex(-55%);
          transform: translatex(-55%);
  width: 30%;
  padding: 12px;
  background-color: #e9e9e9;
  color: #424242;
  border-radius: 16px;
}

.sugesstions .city-item {
  padding: 4px;
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
}

.sugesstions .city-item:hover {
  background-color: white;
}
/*# sourceMappingURL=NavBar.css.map */