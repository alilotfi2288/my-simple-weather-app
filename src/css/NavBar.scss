.nav{
    width: 100%;
    min-height: 10vh;
    background-color: #8E24AA;
    color: white;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 24px;
    h2{
        font-weight: bold;
    }
    .city-search{
        border: none;
        width: 30%;
        padding: 12px;
        border-radius: 24px;
        background-color: rgba(206, 147, 216 , 0.5);
        color: white;
        text-align: center;
        &::placeholder{
            color: white;
        }
        &:focus{
            outline: thin white auto;
        }
    }
    .unit > *{
        padding: 0px 4px;
        font-weight: bold;
        cursor: pointer;
        &:nth-child(2){
            cursor: auto;
        }
    }
}
.sugesstions{
    position: absolute;
    top: 9vh;
    left: 55%;
    transform: translatex(-55%);
    width: 30%;
    padding: 12px;
    background-color: rgb(233, 233, 233);
    color: #424242;
    border-radius: 16px;
    .city-item{
        padding: 4px;
        transition: background-color 0.3s ease-in-out;
        &:hover{
            background-color: white;
        }
    }
}