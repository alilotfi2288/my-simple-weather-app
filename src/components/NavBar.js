import React, { Component } from 'react';
import '../css/NavBar.min.css';
import axios from 'axios';
import { searchConfig } from '../api/weatherConfig';

class NavBar extends Component {
    state = {
        data: []
    }
    handleChange = (e) => {
        if (e.target.value !== '') {
            axios.get(searchConfig(e.target.value))
                .then(res => {
                    this.setState({
                        data: res.data
                    });
                });
        } else {
            this.setState({
                data: []
            });
        }
    }
    handleClick = (city) => {
        this.setState({
            data: []
        });
        this.props.setCity(city);
    }

    render() {
        let { data } = this.state;
        let autoCompleteList = null;
        if (data.length !== 0) {
            autoCompleteList =
                <div className="sugesstions">
                    {data.map(city => {
                        if (city.type === 'city') {
                            return (
                                <div className="city-item" key={Math.random()} onClick={() => { this.handleClick(city.name) }}>
                                    <p>{city.name} {city.country_name}</p>
                                </div>
                            )
                        } else {
                            return null;
                        }
                    })}
                </div>
        }

        return (
            <div className="nav">
                <h2 data-testid="title" className="logo">Weather App</h2>
                <input className="city-search" placeholder="city name..." onChange={this.handleChange} />
                <div className="unit">
                    <span onClick={() => { this.props.changeUnit('C') }}>C</span>
                    <span>|</span>
                    <span onClick={() => { this.props.changeUnit('F') }}>F</span>
                </div>
                {autoCompleteList}

            </div>
        )
    }
}

export default NavBar;