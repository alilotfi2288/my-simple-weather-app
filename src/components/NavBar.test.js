import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

test('title display properly', () => {
    render(<NavBar />);
    let title = screen.getByTestId('title');
    expect(title).toBeInTheDocument();
    expect(title.innerHTML).toBe('Weather App');
});
test('unit spans display properly', () => {
    render(<NavBar />);
    let spans = [screen.getByText('C'), screen.getByText('F')];
    spans.forEach(span => {
        expect(span).toBeInTheDocument();
    });
});
test('input changing test', () => {
    render(<NavBar />);
    let input = screen.getByPlaceholderText('city name...');
    fireEvent.change(input, { target: { value: 'hello' } });
    expect(input.value).toEqual('hello');
});
test('snapShot testing', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
});
