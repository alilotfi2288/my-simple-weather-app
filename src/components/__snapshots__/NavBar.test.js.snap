// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapShot testing 1`] = `
<div
  className="nav"
>
  <h2
    className="logo"
    data-testid="title"
  >
    Weather App
  </h2>
  <input
    className="city-search"
    onChange={[Function]}
    placeholder="city name..."
  />
  <div
    className="unit"
  >
    <span
      onClick={[Function]}
    >
      C
    </span>
    <span>
      |
    </span>
    <span
      onClick={[Function]}
    >
      F
    </span>
  </div>
</div>
`;
